src/utils/diff/parsers/common.ts

```diff
@@ -128,24 +149,27 @@ export abstract class BaseParser {
         return { added, removed }
     }
 
-    abstract match(content: string, options: ParseOptions): RegExpExecArray | null
+    match(content: string): RegExpExecArray | null {
+        return this.buildRegexPattern().exec(content)
+    }
 
     /**
      * Given a string, returns all matches found in the string.
      * Each match includes the variable name and the character range of the
      * matching substring.
      */
-    private getAllMatches(content: string, options: ParseOptions): Match[] {
+    private getAllMatches(content: string): Match[] {
         const matches: Match[] = []
         let cursorPosition = 0
 
         while (cursorPosition < content.length) {
             const substring = content.slice(cursorPosition)
-            const match = this.match(substring, options)
+            const match = this.match(substring)
 
             if (!match) break
 
-            const [matchContent, variableName] = match
+            const [matchContent] = match
+            const variableName = match[1] || match[2] // position or named parameter match
```

Comment:
how come it's a different position in the capture groups depending on the pattern? Also would match[1] be falsey while match[2] is truthy?

Reply:
If a language supports both ordered parameters and named parameters, then the resulting pattern will have two capture groups. Only one of the two will match and the other will be undefined